File: How_To_Program_Exercises/C++How_To_Program_8/Chapter3/TargetHeartRateCalculator/HeartRates.cpp
//
//	HeartRates Class Test
//

#include <iostream>
#include <string>
#include "HeartRates.h"

//Constructor
HeartRates::HeartRates (std::string fName, std::string lName, int d, int m, int y)
{
	firstName = fName;
	lastName = lName;
	birthDay = d;
	birthMonth = m;
	birthYear = y;
}

//Function that sets firstName
void HeartRates::setFirstName(std::string fName)
{
	firstName = fName;
}

//Function that returns firstName
std::string HeartRates::getFirstName()
{
	return firstName;
}

//Function that sets Last Name
void HeartRates::setLastName(std::string lName)
{
	lastName = lName;
}

//Function that returns Last Name
std::string HeartRates::getLastName()
{
	return lastName;
}

//Function that sets the Birthday
void HeartRates::setBirthDay(int d, int m, int y)
{
	birthDay = d;
	birthMonth = m;
	birthYear = y;
}

//Function that returns Age
int HeartRates::getAge()
{
	int dd, mm, yy;
	std::cout << "Enter todays date in the following format d m y: ";
	std::cin >> dd >> mm >> yy;

	age = yy - birthYear;

	//If current date is before birthday, decrement age 
	if (birthMonth > mm)
		age = age - 1;

	if (birthMonth == mm)
		if (birthDay > dd)
			age = age - 1;

	return age;
}

//Function that returns Maximum Heart Rate
int HeartRates::getMaximumHeartRate()
{ 
	return 220 - age;
}

//Function that returns Target Heart Rate
int HeartRates::getTargetHeartRate()
{
	return getMaximumHeartRate() * 675 / 1000;
}

File: How_To_Program_Exercises/JavaHowToProgram9/Chapter3/HeartRateCalculatorClass/HeartRates.java
//
//	HeartRates Class
//	Calculates Maximum and Target Heart Rates
//

class HeartRates
{
	private String firstName;
	private String lastName;
	private int age;	
	private int day;	//Contains birth date
	private int month;	//Contains birth month 
	private int year;	//Contains birth year
	
	//Constructor
	public HeartRates (String fName, String lName, int d, int m, int y)
	{
		firstName = fName;
		lastName = lName;
		day = d;
		month = m;
		year = y;
	}
	
	//Returns the firstName field
	public String getFirstName ()
	{
		return firstName;
	}
	
	//Sets the firstName field
	public void setFirstName (String fName)
	{
		firstName = fName;
	}
	
	//Returns field lastName
	public String getLastName ()
	{
		return lastName;
	}
	
	//Sets lastName field
	public void setLastName (String lName)
	{
		lastName = lName;
	}
	
	//Sets date of birth fields
	public void setDateOfBirth (int d, int m, int y)
	{
		day = d;
		month = m;
		year = y;
	}
	
	//Calculates and return the age provided current date
	public int getAge ( int d, int m, int y)
	{
		age = y - year;
		
		if ( month > m )
			age = age - 1;
			
		if ( month == m )
			if ( day > d )
				age = age - 1;
		
		return age;
	}
	
	//Calculates and returns maximum heart rate
	public int getMaximumHeartRate ()
	{
		return 220 - age;
	}
	
	//Calculates and return target heart rate
	public int getTargetHeartRate ()
	{
		return getMaximumHeartRate() * 675 / 1000;
	}
}